tools/clang/tools/dxv/dxv.cpp

```diff
@@ -80,19 +106,50 @@ void DxvContext::Validate() {
       std::string msg(pStart);
       IFTMSG(status, msg);
     } else {
+      // Source was unsigned DxilContainer, write signed container if it's now signed:
+      if (bSourceIsDxilContainer && bSourceIsUnsigned) {
+        const hlsl::DxilContainerHeader *pHeader = hlsl::IsDxilContainerLike(
+            pSource->GetBufferPointer(), pSource->GetBufferSize());
+        hlsl::DxilContainerHash blankHash = {};
+        if (memcmp(&pHeader->Hash, &blankHash,
+                   sizeof(hlsl::DxilContainerHash)) != 0) {
+          if (OutputFilename.empty()) {
+            OutputFilename.assign(InputFilename);
```

Comment:
So this rewrites the input container with a signature if you don't specify an output? That might come as a surprise. Probably fine if the file is writable. If not, it might fail when someone only wants to know if the dxil is valid.

Reply:
True, perhaps we only output if `-o` specified, and allow -o to point to input file to overwrite it.